### PR TITLE
Switch from node sass to sass

### LINKED ISSUE #170: Use sass instead of node-sass
- Switch to [sass](https://www.npmjs.com/package/sass) instead of `node-sass`
- Use `node-slim` instead of `node`. It was changed here: https://github.com/canonical-web-and-design/juju.is/pull/166/files

### PR DESCRIPTION
**Pull request by steverydz**
## Done
- Changed from node-sass to sass
- Changed from sass-lint to stylelint

## QA
- Lint the styles: `yarn lint-scss`
- Make sure the CSS builds: `yarn build-css`
- Run the site and check that it doesn't look broken

## Issue
Fixes #170

### CODE REVIEW

**Comment by sowasred2012** [comment]
@bartaz are we definitely swatching to dart-sass on Vanilla? 

Running `dotrun build-css` raises a few warnings, mostly about the [slash divider operator](https://sass-lang.com/documentation/breaking-changes/slash-div) being deprecated and removed in the 2.0 release of dart-sass, but all but one of the warnings relate to Vanilla SCSS, e.g:

Screenshot from 2021-05-25 17-06-05

If we merge this, doesn't that mean Vanilla is locked into using dart-sass (or, that any project that does use dart-sass is forced to stay on versions earlier 2.0)?

**Comment by bartaz** [comment]
@sowasred2012  node-sass is deprecated, so we need to eventually more to dart sass anyway. Currently we support dart-sass (meaning Vanilla should build without errors in dart-sass), but by default we build in node-sass.

I think in 3.0 we should drop support for node sass, and fix any issues that cause dart-sass incompatibilities.

So, to summarize, it's OK for any project to use dart sass to build. Ignore warnings, but ping us if there are any errors. We do check this on our CI, so it should build in most cases.

File: static/sass/_patterns_docs.scss [hunk 1]
```diff
@@ -24,42 +24,6 @@
...
-
-  .p-sidenav__toggle {
-    background-size: $sp-medium;
-    cursor: pointer;
-    float: right;
-    padding: $sp-x-large;
-  }
-
   .p-sidenav__body {
     padding: $sp-large / 2 $sp-large;
```
Comment (sowasred2012) [inline static/sass/_patterns_docs.scss:28]: This raises a warning about the [slash divider operator being deprecated](https://sass-lang.com/documentation/breaking-changes/slash-div), and removed in the next major version - could use the new syntax here?

```suggestion
    padding: math.div($sp-large, 2) $sp-large;
```

You'd also need to add `@use "sass:math";` at the top of the file.

**Review by sowasred2012 — APPROVED** [review]
LGTM, just one suggestion

**Pull request merged**